src/app/ui/shared/components/studio/section/helpers/createPlotDataObject.ts

```diff
@@ -0,0 +1,78 @@
+import { Dash, Data } from 'plotly.js-dist-min';
+import { GetAllPhasesParams, PlotObjectType } from './types';
+
+const getColor = (
+  type: PlotObjectType
+): {
+  color: string;
+  dash: Dash;
+} => {
+  if (type === 'span') return { color: 'red', dash: 'solid' };
+  if (type === 'support') return { color: 'blue', dash: 'solid' };
+  if (type === 'insulator') return { color: 'green', dash: 'solid' };
+  return { color: 'black', dash: 'solid' };
+};
+
+const getMode = (
+  type: PlotObjectType
+):
+  | 'number'
+  | 'text'
+  | 'delta'
+  | 'gauge'
+  | 'none'
+  | 'lines'
+  | 'markers'
+  | 'lines+markers'
+  | 'text+markers'
+  | 'text+lines'
+  | 'text+lines+markers'
+  | 'number+delta'
+  | 'gauge+number'
+  | 'gauge+number+delta'
+  | 'gauge+delta'
+  | undefined => {
+  if (type === 'support') return 'text+lines';
+  return 'lines';
+};
```

Comment:
I don't get why there's so much types here. Is it for later use?

```suggestion
const getMode = (type: PlotObjectType) => {
  return type === 'support' ? 'text+lines' : 'lines';
};
```